### DISCUSSION TITLE
Simple questions

### DISCUSSION THREAD

**Discussion by xflou** [PowerShell/PowerShell#17487] [General]
Hello and thanks in advance!

Running Powershell on Rocky Linux 8. 

Powershell Version:
Name                           Value
**PSVersion                      7.2.4
PSEdition                      Core
GitCommitId                    7.2.4
OS                             Linux 4.18.0-348.el8.0.2.x86_64 #1 SMP Sun Nov 14 00:51:12 UTC 2021
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0**

1. Trying to find location for the list of commands supported by 7.2.4 
2. Trying to use the following commands to access and or add information to AD:
Get-ADGroupMember -Server “ServerName” -identity “GroupName” -Recursive | select name,objectclass,displayname
Add-ADGroupMember -Identity “GroupName” -Members “AcctName” -Credential “PrivAcctName”

**However, getting the following errors:**
`
Get-ADGroupMember: The term 'Get-ADGroupMember' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
`
`PS /> import-module activedirectory
Import-Module: The specified module 'activedirectory' was not loaded because no valid module file was found in any module directory.
`

Thanks again!

**Comment by kasini3000** [comment]
### from linux client ,login to windwos server ,then send command and run

1 install omi   https://github.com/microsoft/omi
or https://github.com/jborean93/omi

2 k_run_ip_port_l2w_user_pwd.ps1 -ip your_windows_server_ip -port xxx -user xxx -pwd xxx {
Get-ADGroupMember -Server “ServerName” -identity “GroupName” -Recursive | select name,objectclass,displayname
Add-ADGroupMember -Identity “GroupName” -Members “AcctName” -Credential “PrivAcctName”

}

After install microsoft OMI or jborean93/omi , then you can connect from linux to windows by WINRM-5985 (ntlm).
but OMI has some bug ,sctipt file name on windows does not support characters other than English.
you can use this script:
https://gitee.com/chuanjiao10/kasini3000/blob/master/k_run_ip_port_l2w_user_pwd.ps1

----

[kasini3000](https://gitee.com/chuanjiao10/kasini3000) site mirror:(https://github.com/kasini3000/kasini3000)   
win,linux devops automation batch script framework.(It is similar to Puppet，Ansible，pipeline) 
Open source, free, cross-platform 
English manual: https://gitee.com/chuanjiao10/kasini3000/blob/master/docs/kasini3000_manual.xlsx

[kasini3000_agent_linux](https://gitee.com/chuanjiao10/kasini3000_agent_linux) Shell script,one click install powershell on linux,modify sshd_config for PsRemote.
